packages/vue-components/src/Navbar.vue

```diff
@@ -77,10 +77,41 @@
       toggleCollapse (e) {
         e && e.preventDefault()
         this.collapsed = !this.collapsed
+      },
+      highlightCurrentPage() {
+        function hasMatchingUrl(elements, currPage) {
+          if (!elements || !elements.children) {
+            return false;
+          }
+          // Only check <a> leaf nodes
+          if (elements.children.length === 0) {
```

Comment:
hmm, a query selector on a class might not work since dropdown menus in nav bars only have a class `dropdown-item` whereas ordinary navbar links have a class of `nav-link`. So if we want highlighting on dropdown links, another document.query doesn't seem to be possible, hence the class-agnostic DFS

![image](https://user-images.githubusercontent.com/38814428/97794386-364fab00-1c34-11eb-86e0-27d455b23042.png)

Reply:
not on the class, but the anchor links you're checking here `'a[href]'`